pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -345,8 +346,14 @@ def _process_training_step_output(self, training_step_output: STEP_OUTPUT) -> Op
         if isinstance(training_step_output, dict):
             loss = training_step_output.pop("loss", None)
             hiddens = training_step_output.pop("hiddens", None)
+
             if hiddens is not None:
```

Comment:
using `apply_to_collection`, now the None check is not necessary anymore, as it will compare the type anyway. Can you remove it?